### PR TITLE
Removing unecessary early stopping calls

### LINKED ISSUE #1751: Early Stopping behavior
Hi there,
thanks for the great library (I am using 0.7.5.). I am not following the bug report template as I'm not sure this is indeed a bug, or simply I cannot understand how early stopping is implemented. My code looks as follows:

        early_stop_callback = EarlyStopping(
            monitor='val_acc',
            min_delta=0.0,
            patience=80,
            verbose=True,
            mode=self.mode
        )

        trainer = Trainer(
            early_stop_callback=early_stop_callback,
            auto_select_gpus=True,
            max_epochs=200,
            terminate_on_nan=True,
            show_progress_bar=True,
            fast_dev_run=False,
            gpus=1
        )

As I understand it, the model should perform early stopping after AT LEAST 80 epochs have passed without improvement on the validation accuracy. However, in my case, early stopping happened at epoch 75. Is this how it should be?

As I said, I am not sure this is actually a bug or a choice (perhaps early stopping is implemented at the batch level?). If it is indeed a bug, I will work a reproducible example. Thank you!

### PR DESCRIPTION
**Pull request by mateuszpieniak**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes #1751 

## PR review    
The removed early stopping calls are conditioned in different cases. 
* The first one is called right after the validation dataset evaluation. 
* The second one is in case of no validation dataset i.e. no `validation_step` implemented. 

It's enough to call it once after the training because `self.early_stop_callback.on_epoch_end` can read required metrics from `trainer.callback_metrics`.

### CODE REVIEW

**Comment by williamFalcon** [comment]
i think there was a reason for both of those. need to think about this more

**Comment by jeremyjordan** [comment]
@williamFalcon Directly calling the callback methods (eg. `on_epoch_end`) is an anti-pattern, we should delegate that responsibility to the callback handler. When a user is writing callback code for `on_epoch_end` they should be able to trust that it will only be called at the end of an epoch.
